Lib/xml/dom/pulldom.py

```diff
@@ -293,41 +298,46 @@ def clear(self):
         self.parser = None
         self.stream = None
 
+
 class SAX2DOM(PullDOM):
+    def __init__(self, documentFactory=None, handle_sax=True):
+        super().__init__(documentFactory, handle_sax)
 
-    def startElementNS(self, name, tagName , attrs):
-        PullDOM.startElementNS(self, name, tagName, attrs)
+    def startElementNS(self, name, tagName, attrs):
+        super().startElementNS(name, tagName, attrs)
         curNode = self.elementStack[-1]
         parentNode = self.elementStack[-2]
         parentNode.appendChild(curNode)
 
     def startElement(self, name, attrs):
-        PullDOM.startElement(self, name, attrs)
+        super().startElement(name, attrs)
         curNode = self.elementStack[-1]
         parentNode = self.elementStack[-2]
         parentNode.appendChild(curNode)
 
     def processingInstruction(self, target, data):
-        PullDOM.processingInstruction(self, target, data)
-        node = self.lastEvent[0][1]
-        parentNode = self.elementStack[-1]
-        parentNode.appendChild(node)
+        super().processingInstruction(target, data)
+        if self.lastEvent[0] is not None:
```

Comment:
Was this a bug that is now fixed? Is there a test that exercises this case?

Reply:
This is the core of the fix for this issue. A test case already exists and removing the expected failure flag means that the PI test before root element will now be reported as successful when it succeeds. No new test case is needed.

Reply:
Ah, that makes sense, thanks for the clarification.